code/modules/mob/living/carbon/xenomorph/Powers.dm

```diff
@@ -99,6 +99,19 @@
 	if(RC.build_overlay_icon)
 		warning = new RC.build_overlay_icon(current_turf)
 
+	if(RC.build_animation_effect)
+		warning = new RC.build_animation_effect(current_turf)
+
+		if(wait_time != 2 SECONDS)
+
+			if(wait_time == 1 SECONDS)
+				warning.icon_state = "[warning.icon_state]Fast"
+
+			if(wait_time == 4 SECONDS)
+				warning.icon_state = "[warning.icon_state]Slow"
```

Comment:
switch statement for wait_time instead of if chain

Reply:
Done so.